__tests__/inlineStyles.test.js

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
```

Comment:
I agree that it would be better. At first I thought that would be easy but typescript declarations in `react-native-reanimated.d.ts` and in reanimated code (`timing.ts`, `utils.ts`, `repeat.ts` and so on) are different. Tests use declarations from the code and in my opinion the types in the code don't make sense. Like why repeat animation takes repeat animation as an argument? It can be any animation.
```ts
export function withRepeat(
  _nextAnimation: NextAnimation<RepeatAnimation>,
  numberOfReps = 2,
  reverse = false,
  callback?: AnimationCallback
): Animation<RepeatAnimation>
```
I tried to update those types but it goes deeper and deeper. You have this monstrosity in `utils.ts`:
```ts
type AnimationToDecoration<T extends AnimationObject | StyleLayoutAnimation> =
  T extends StyleLayoutAnimation
    ? Record<string, unknown>
    : T extends DelayAnimation
    ? NextAnimation<DelayAnimation>
    : T extends RepeatAnimation
    ? NextAnimation<RepeatAnimation>
    : T extends SequenceAnimation
    ? NextAnimation<SequenceAnimation>
    : AnimatableValue | T;

export function defineAnimation<
  T extends AnimationObject | StyleLayoutAnimation
>(
  starting: AnimationToDecoration<T>,
  animationFunctionCall: AnimationFunctionCall,
  factory: () => T
): T {
```
`starting` doesn't have to be an animation, it may be `AnimatableValue`. `defineAnimation` can return animation, animation factory or a value depending on where is it called and it's already `// @ts-ignore`'d
Correcting those types should be done in separate PR, this one is already large and those types were wrong before inline styles.